Formula/xserver.rb

```diff
@@ -0,0 +1,114 @@
+class Xserver < Formula
+  desc "X Window System display server"
+  homepage "https://www.x.org"
+  url "https://gitlab.freedesktop.org/xorg/xserver.git",
+    tag:      "xorg-server-21.1.3",
+    revision: "85397cc2efe8fa73461cd21afe700829b2eca768"
+  license all_of: ["MIT", "APSL-2.0"]
+
+  depends_on "font-util"   => :build
+  depends_on "libxkbfile"  => :build
+  depends_on "meson"       => :build
+  depends_on "ninja"       => :build
+  depends_on "pkg-config"  => :build
+  depends_on "util-macros" => :build
+  depends_on "xorgproto"   => :build
+  depends_on "xtrans"      => :build
+
+  depends_on "libapplewm"
+  depends_on "libxfixes"
+  depends_on "libxfont2"
+  depends_on "mesa"
+  depends_on "pixman"
+  depends_on "xcb-util"
+  depends_on "xcb-util-image"
+  depends_on "xcb-util-keysyms"
+  depends_on "xcb-util-renderutil"
+  depends_on "xcb-util-wm"
+  depends_on "xkbcomp"
+  depends_on "xkeyboardconfig"
+
+  on_linux do
+    depends_on "libdrm"
+    depends_on "libepoxy"
+    depends_on "libxcvt"
+    depends_on "libxshmfence"
+    depends_on "systemd"
+  end
+
+  def install
+    # align with formula `font-util`
+    font_path = %w[misc TTF OTF Type1 100dpi 75dpi].map do |p|
+      HOMEBREW_PREFIX/"share/fonts/X11"/p
+    end
+    if OS.mac?
+      font_path += %W[
+        #{HOMEBREW_PREFIX}/share/system_fonts
+        /Library/Fonts
+        /System/Library/Fonts
+      ]
+    end
+    meson_args = std_meson_args + %W[
+      -Dxephyr=true
+      -Dxf86bigfont=true
+      -Dxcsecurity=true
+
+      -Dmodule_dir=#{HOMEBREW_PREFIX}/lib/xorg/modules
+      -Ddefault_font_path=#{font_path.join ","}
+      -Dxkb_dir=#{HOMEBREW_PREFIX}/share/X11/xkb
+      -Dxkb_bin_dir=#{Formula["xkbcomp"].opt_bin}
+      -Dxkb_output_dir=#{HOMEBREW_PREFIX}/X11/xkb/compiled
+
+      -Dbundle-id-prefix=#{Formula["xinit"].plist_name.chomp ".startx"}
+      -Dbuilder_addr=#{tap.remote}
+      -Dbuilder_string=#{tap.name}
+    ]
+    # macOS doesn't provide `authdes_cred` so `secure-rpc=false`
+    # glamor needs GLX enabled `libepoxy` on macOS
+    if OS.mac?
+      meson_args += %W[
+        -Dsecure-rpc=false
+        -Dapple-applications-dir=#{libexec}
+      ]
+    end
+
+    system "meson", "build", *meson_args
+    system "meson", "compile", "-C", "build"
+    system "meson", "install", "-C", "build"
+    bin.install_symlink bin/"Xquartz" => "X" if OS.mac?
+  end
+
+  def post_install
+    if OS.mac?
+      system "/System/Library/Frameworks/CoreServices.framework"\
+             "/Frameworks/LaunchServices.framework/Support/lsregister",
+             "-R", "-f", libexec/"X11.app"
```

Comment:
Does this spam the registry with different Cellar paths or are they trimmed automatically without issue? I wonder if it makes sense to use opt_libexec.

Reply:
Output of `lsregister -dump`

Use `libexec`:
```
bundle id:                  X11 (0x8aac)
class:                      kLSBundleClassApplication (0x2)
container:                  / (0x4)
mount state:                mounted
sequenceNum:                35500
path:                       /usr/local/Cellar/xserver/21.1.3/libexec/X11.app (0xb8b0)
directory:                  Other (255)
name:                       X11
localizedShortNames:        "ar" = ?, "ca" = ?, "cs" = ?, "da" = ?, "Dutch" = ?, "el" = ?, "English" = ?, "fi" = ?, "French" = ?, "German" = ?, "he" = ?, "hr" = ?, "hu" = ?, "Italian" = ?, "Japanese" = ?, "ko" = ?, "LSDefaultLocalizedValue" = "X11", "no" = ?, "pl" = ?, "pt" = ?, "pt_PT" = ?, "ro" = ?, "ru" = ?, "sk" = ?, "Spanish" = ?, "sv" = ?, "th" = ?, "tr" = ?, "uk" = ?, "zh_CN" = ?, "zh_TW" = ?
category:                   public.app-category.utilities (0x19d4c)
identifier:                 homebrew.mxcl.X11
canonical id:               homebrew.mxcl.x11
version:                    12101003.0 ({length = 32, bytes = 0x8ba5b800 00000000 00000000 00000000 ... 00000000 00000000 })
versionString:              12101003
displayVersion:             21.1.3
mod date:                   2022-01-03 06:23 (POSIX 1641162210, 𝛥 4mths 3wks 3days 4hr 6min 31sec)
reg date:                   2022-05-20 17:47 (POSIX 1653040070, 𝛥 6days 16hr 42min 11sec)
rec mod date:               2022-05-20 17:47 (POSIX 1653040070, 𝛥 6days 16hr 42min 11sec)
type code:                  'APPL' (4150504c)
creator code:               'x11a' (78313161)
bundle flags:               shell-script (0000000000000010)
icon flags:                 relative-icon-path (0000000000000001)
slices:                     x86_64 (0000000000000008)
item flags:                 package  application  container  native-app  extension-hidden (000000000010008e)
App Nap:                    capable  can-change (0000000000000005)
eGPU:                       capable  can-change (0000000000000005)
safeAperture system fullscreen: capable  can-change (0000000000000005)
safeAperture app fullscreen: capable  can-change (0000000000000005)
safeAperture windowed:      capable  can-change (0000000000000005)
platform:                   #
iconDict:                   1 values (143604 (0x230f4))
                            {
                                CFBundlePrimaryIcon =     {
                                    CFBundleIconFile = "X11.icns";
                                };
                            }
icons:                      Contents/Resources/X11.icns
executable:                 Contents/MacOS/X11
inode:                      8868630726
exec inode:                 8868630730
infoDictionary:             15 values (143608 (0x230f8))
                            {
                                CFBundleDevelopmentRegion = English;
                                CFBundleExecutable = X11;
                                CFBundleGetInfoString = "homebrew.mxcl.X11";
                                CFBundleIconFile = "X11.icns";
                                CFBundleIdentifier = "homebrew.mxcl.X11";
                                CFBundleInfoDictionaryVersion = "6.0";
                                CFBundleName = X11;
                                CFBundlePackageType = APPL;
                                CFBundleShortVersionString = "21.1.3";
                                CFBundleSignature = x11a;
                                CFBundleVersion = 12101003;
                                NSHumanReadableCopyright = "\U00a9 2003-2021 Apple Inc.\n\U00a9 2003 XFree86 Project, Inc.\n\U00a9 2003-2021 X.org Foundation, Inc.\n";
                                NSMainNibFile = main;
                                NSPrincipalClass = X11Application;
                                NSSupportsAutomaticGraphicsSwitching = 1;
                            }
activityTypes:              NOTIFICATION#:homebrew.mxcl.X11, pv-f222b0b49ef1d7
```

Use `opt_libexec`:
```
bundle id:                  X11 (0x9190)
class:                      kLSBundleClassApplication (0x2)
container:                  / (0x4)
mount state:                mounted
sequenceNum:                37264
path:                       /usr/local/Cellar/xserver/21.1.3/libexec/X11.app (0xc008)
directory:                  Other (255)
name:                       X11
localizedShortNames:        "ar" = ?, "ca" = ?, "cs" = ?, "da" = ?, "Dutch" = ?, "el" = ?, "English" = ?, "fi" = ?, "French" = ?, "German" = ?, "he" = ?, "hr" = ?, "hu" = ?, "Italian" = ?, "Japanese" = ?, "ko" = ?, "LSDefaultLocalizedValue" = "X11", "no" = ?, "pl" = ?, "pt" = ?, "pt_PT" = ?, "ro" = ?, "ru" = ?, "sk" = ?, "Spanish" = ?, "sv" = ?, "th" = ?, "tr" = ?, "uk" = ?, "zh_CN" = ?, "zh_TW" = ?
category:                   public.app-category.utilities (0x19d4c)
identifier:                 homebrew.mxcl.X11
canonical id:               homebrew.mxcl.x11
version:                    12101003.0 ({length = 32, bytes = 0x8ba5b800 00000000 00000000 00000000 ... 00000000 00000000 })
versionString:              12101003
displayVersion:             21.1.3
mod date:                   2022-01-03 06:23 (POSIX 1641162210, 𝛥 4mths 3wks 3days 4hr 8min 39sec)
reg date:                   2022-05-27 10:31 (POSIX 1653618704, 𝛥 25sec)
rec mod date:               2022-05-27 10:31 (POSIX 1653618704, 𝛥 25sec)
type code:                  'APPL' (4150504c)
creator code:               'x11a' (78313161)
bundle flags:               shell-script (0000000000000010)
icon flags:                 relative-icon-path (0000000000000001)
slices:                     x86_64 (0000000000000008)
item flags:                 package  application  container  native-app  extension-hidden (000000000010008e)
App Nap:                    capable  can-change (0000000000000005)
eGPU:                       capable  can-change (0000000000000005)
safeAperture system fullscreen: capable  can-change (0000000000000005)
safeAperture app fullscreen: capable  can-change (0000000000000005)
safeAperture windowed:      capable  can-change (0000000000000005)
platform:                   #
iconDict:                   1 values (148276 (0x24334))
                            {
                                CFBundlePrimaryIcon =     {
                                    CFBundleIconFile = "X11.icns";
                                };
                            }
icons:                      Contents/Resources/X11.icns
executable:                 Contents/MacOS/X11
inode:                      8868630726
exec inode:                 8868630730
infoDictionary:             15 values (148280 (0x24338))
                            {
                                CFBundleDevelopmentRegion = English;
                                CFBundleExecutable = X11;
                                CFBundleGetInfoString = "homebrew.mxcl.X11";
                                CFBundleIconFile = "X11.icns";
                                CFBundleIdentifier = "homebrew.mxcl.X11";
                                CFBundleInfoDictionaryVersion = "6.0";
                                CFBundleName = X11;
                                CFBundlePackageType = APPL;
                                CFBundleShortVersionString = "21.1.3";
                                CFBundleSignature = x11a;
                                CFBundleVersion = 12101003;
                                NSHumanReadableCopyright = "\U00a9 2003-2021 Apple Inc.\n\U00a9 2003 XFree86 Project, Inc.\n\U00a9 2003-2021 X.org Foundation, Inc.\n";
                                NSMainNibFile = main;
                                NSPrincipalClass = X11Application;
                                NSSupportsAutomaticGraphicsSwitching = 1;
                            }
activityTypes:              NOTIFICATION#:homebrew.mxcl.X11, pv-f222b0b49ef1d7
```
 
It seems always using the real path, and `lsregister` can update the item when it already exists.

Reply:
Thanks for checking!